clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12519,16 +12519,13 @@ def err_builtin_is_within_lifetime_invalid_arg : Error<
   "%select{non-|function }0pointer argument to '__builtin_is_within_lifetime' "
   "is not allowed">;
 
-def err_builtin_invalid_arg_type: Error <
-  "%ordinal0 argument must be "
-  "%select{a vector, integer or floating point type|a matrix|"
-  "a pointer to a valid matrix element type|"
-  "a signed integer or floating point type|a vector type|"
-  "a floating point type|"
-  "a vector of integers|"
-  "an unsigned integer|"
-  "an 'int'|"
-  "a vector of floating points}1 (was %2)">;
+def err_builtin_invalid_arg_type: Error<
+  "%ordinal0 argument must be a"
+  "%select{| scalar| vector| matrix| vector,| vector of| scalar or vector of}1"
+  "%select{| integer| signed integer| unsigned integer| 'int'|"
```

Comment:
```suggestion
  "%ordinal0 argument must be a "
  "%select{|scalar|vector|matrix|vector,|vector of|scalar or vector of}1"
  "%select{|integer|signed integer|unsigned integer|'int'|"
```
(Moved some whitespace around.)

That stray comma really bugs me because that can lead to diagnostics like:

`1st argument must be a vector, 'int' type (was 'blah')`

This is a bit of a hard diagnostic to reason about, but I don't think it's worth trying to split it up more, either.

Reply:
I don't insist on a change, I think what you've got is complex^Wgood enough. :-)

Reply:
Thanks for the review. I've pushed the version I outlined above but included detailed comments around each component, to hopefully help. I can revert this latest commit if desired.